backend-services/user-verification-service/src/main/java/org/cardano/foundation/voting/client/KeriVerificationClient.java

```diff
@@ -0,0 +1,148 @@
+        requestBody.put("payload", payload);
+
+        HttpEntity<Map<String, String>> entity = new HttpEntity<>(requestBody, headers);
+
+        try {
+            ResponseEntity<String> response = restTemplate.exchange(url, HttpMethod.POST, entity, String.class);
+
+            if (response.getStatusCode().is2xxSuccessful()) {
+                return Either.right(true);
+            } else {
+                return Either.left(Problem.builder()
+                        .withTitle("KERI_VERIFICATION_FAILED")
+                        .withDetail("The Keri-specific condition was not met.")
+                        .withStatus(new HttpStatusAdapter(response.getStatusCode()))
+                        .build());
+            }
+        } catch (HttpClientErrorException e) {
+
+            return Either.left(Problem.builder()
+                    .withTitle("KERI_VERIFICATION_ERROR")
+                    .withDetail("Unable to verify signature, reason: " + e.getMessage())
+                    .withStatus(new HttpStatusAdapter(e.getStatusCode()))
+                    .build());
+        }
+    }
+
+    public Either<Problem, Boolean> registerOOBI(String oobi) {
+        String url = String.format("%s/oobi", keriVerifierBaseUrl);
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.add("Content-Type", "application/json");
+
+        Map<String, String> requestBody = new HashMap<>();
+        requestBody.put("oobi", oobi);
+
+        HttpEntity<Map<String, String>> entity = new HttpEntity<>(requestBody, headers);
+
+        try {
+            ResponseEntity<String> response = restTemplate.exchange(url, HttpMethod.POST, entity, String.class);
+
+            if (response.getStatusCode().is2xxSuccessful()) {
+                return Either.right(true);
+            } else {
+                return Either.left(Problem.builder()
+                        .withTitle("OOBI_REGISTRATION_FAILED")
+                        .withDetail("Failed to register OOBI.")
+                        .withStatus(new HttpStatusAdapter(response.getStatusCode()))
+                        .build());
+            }
+        } catch (HttpClientErrorException e) {
+            return Either.left(Problem.builder()
+                    .withTitle("OOBI_REGISTRATION_ERROR")
+                    .withDetail("Unable to register OOBI, reason: " + e.getMessage())
+                    .withStatus(new HttpStatusAdapter(e.getStatusCode()))
+                    .build());
+        }
+    }
+
+    public Either<Problem, String> getOOBI(String oobi, Integer maxAttempts) {
+        String url = String.format("%s/oobi?url=%s", keriVerifierBaseUrl, oobi);
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.add("Content-Type", "application/json");
+
+        HttpEntity<Void> entity = new HttpEntity<>(headers);
+
+        int attempts = (maxAttempts == null) ? 1 : maxAttempts;
+        int attempt = 0;
+
+        while (attempt < attempts) {
```

Comment:
Why do wee need to hammer this service?

Reply:
Sometimes Keria takes time to handle and finally persist an action like register an oobi or get an ACDC. We have something similar in IDW here like `waitForAcdcToAppear()`

https://github.com/cardano-foundation/cf-identity-wallet/blob/dc21a9da0236ad2c79a9aa5030846b8a499e93c7/src/core/agent/services/ipexCommunicationService.ts#L180